webapp/javascript/components/FlameGraph/FlameGraphComponent/index.tsx

```diff
@@ -139,7 +142,18 @@ export default function FlameGraphComponent(props: FlamegraphProps) {
           ExportData={ExportData}
         />
 
-        <div>
+        {dataUnavailable ? (
+          <div className="error-message">
+            <span>
+              No profiling data available for this application / time range.
+            </span>
+          </div>
+        ) : null}
+        <div
+          style={{
+            opacity: dataUnavailable ? 0 : 1,
```

Comment:
I'm a bit curious, why was this needed, as opposed to simply not rendering the canvas?

Reply:
Because of the logic attached to `canvasRef`. I imagine simply not rendering the canvas is possible, though it would require some larger refactor across different components.